Paymetheus/ViewModels/PurchaseTicketsViewModel.cs

```diff
@@ -0,0 +1,404 @@
+            }
+        }
+
+        private int _blocksToRetarget;
+        public int BlocksToRetarget
+        {
+            get { return _blocksToRetarget; }
+            internal set { _blocksToRetarget = value; RaisePropertyChanged(); }
+        }
+
+        private void PurchaseTicketsAction()
+        {
+            var shell = ViewModelLocator.ShellViewModel as ShellViewModel;
+            if (shell != null)
+            {
+                var _account = SelectedAccount.Account;
+                var _spendLimit = StakeDifficultyProperties.Price;
+                int _requiredConfirms = 2; // TODO allow user to set
+                uint _expiryHeight = (uint)_expiry + (uint)StakeDifficultyProperties.Height;
+
+                Amount _splitFeeLocal = 0;
+                Amount _ticketFeeLocal = 0;
+                if (_feesChecked)
+                {
+                    _splitFeeLocal = _splitFee;
+                    _ticketFeeLocal = _ticketFee;
+                }
+
+                Func<string, Task<bool>> action =
+                    passphrase => PurchaseTicketsWithPassphrase(passphrase, _account, _spendLimit, 
+                    _requiredConfirms, _ticketAddress, (uint)_number, _poolAddress, _poolFees,
+                    _expiryHeight, _splitFeeLocal, _ticketFeeLocal);
+                shell.VisibleDialogContent = new PassphraseDialogViewModel(shell, 
+                    "Enter passphrase to purchase tickets", 
+                    "PURCHASE", 
+                    action);
+            }
+        }
+
+        private string _responseString = "";
+        public string ResponseString
+        {
+            get { return _responseString; }
+            set { _responseString = value; RaisePropertyChanged(); }
+        }
+
+        private async Task<bool> PurchaseTicketsWithPassphrase(string passphrase, Account account, 
+            Amount spendLimit, int reqConfs, Address ticketAddress, uint number, Address poolAddress,
+            double poolFees, uint expiry, Amount txFee, Amount ticketFee)
+        {
+            List<Blake256Hash> purchaseResponse;
+            var walletClient = App.Current.Synchronizer.WalletRpcClient;
+            try
+            {
+                purchaseResponse = await walletClient.PurchaseTicketsAsync(account, spendLimit, 
+                    reqConfs, ticketAddress, number, poolAddress, poolFees, expiry, txFee,
+                    ticketFee, passphrase);
+            }
+            catch (RpcException ex) when (ex.Status.StatusCode == StatusCode.InvalidArgument)
+            {
+                ResponseString = "Invalid argument error: " + ex.ToString();
+                return true;
+            }
+            catch (RpcException ex) when (ex.Status.StatusCode == StatusCode.FailedPrecondition)
+            {
+                ResponseString = "Failed precondition error: " + ex.ToString();
+                return true;
+            }
+            catch (Exception ex)
```

Comment:
all other errors not explicitly handled above are unexpected and we should let the global exception handler deal with it.